### PR TITLE
Feature/user class

### PR DESCRIPTION
**Pull request by Eleanorgruth**
This branch includes:

- Unit tests for User class 
- Code for User class

Worked: Eleanor & Lauren
 
Please look at:
- Our unit tests and make sure that we are testing happy and sad paths, it might helpful be when reviewing this to refer to the rubric and the testing section :)

### CODE REVIEW

File: test/User-test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  });
+  it('should have a list of recipes to cook', () => {
+    expect(user.recipesToCook).to.deep.equal([])
+  });
+  it('should have a method to add recipes to cook', () => {
```
Comment (jmyacobn) [inline test/User-test.js:34]: Does this add a recipe if it is already there? If not, should we add a conditional in the method? If so, should we test that as our sad path here?

**Review by jmyacobn — APPROVED** [review]
Eleanor and Lauren, your user class and unit tests look good. See my three questions/comments for consideration. Two are connected as they impact the user class and its unit test. The other is a wondering for our unit tests.

File: test/User-test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  });
+  it('should have a list of recipes to cook', () => {
```
Comment (knpausch) [inline test/User-test.js:31]: optional rephrasing: "should initialize list of recipes to an empty array"

File: test/User-test.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    expect(user.recipesToCook).to.deep.equal([recipe1, recipe2])
+  });
+  it('should have a method to remove recipes to cook', () => {
+    user.addRecipesToCook(recipe1)
+    user.addRecipesToCook(recipe2)
+    expect(user.recipesToCook).to.deep.equal([recipe1, recipe2])
+    user.removeRecipesToCook(recipe2)
+    expect(user.recipesToCook).to.deep.equal([recipe1])
+    user.removeRecipesToCook(recipe1)
+    expect(user.recipesToCook).to.deep.equal([])
```
Comment (knpausch) [inline test/User-test.js:47]: great set of expect tests here! Tested all scenarios

File: test/User-test.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    expect(user.filterToCookByTag("hor d'oeuvre")).to.deep.equal([recipe1])
+    expect(user.filterToCookByTag("main dish")).to.deep.equal([recipe2])
+    expect(user.filterToCookByTag("munchies")).to.deep.equal([])
+  })
+  it('should filter recipesToCook by name', () => {
+    user.addRecipesToCook(recipe1)
+    user.addRecipesToCook(recipe2)
+    expect(user.filterToCookByName("Loaded Chocolate Chip Pudding Cookie Cups")).to.deep.equal([recipe1])
+    expect(user.filterToCookByName("Maple Dijon Apple Cider Grilled Pork Chops")).to.deep.equal([recipe2])
+    expect(user.filterToCookByName("suuushhhhhhiiii")).to.deep.equal([])
```
Comment (knpausch) [inline test/User-test.js:61]: haha great sad test here

**Review by knpausch — APPROVED** [review]
Great work. I agree with the comments Jen left, seeing if we could avoid adding a duplicate recipe. Great unit testing scenarios

**Pull request merged**